pandas/core/apply.py

```diff
@@ -465,6 +452,7 @@ def agg_or_apply_dict_like(
         assert op_name in ["agg", "apply"]
 
         obj = self.obj
+        assert isinstance(obj, (ABCSeries, ABCDataFrame))
```

Comment:
EDIT: if your idea is to make `Apply.obj` only be `Series | DataFrame` then there's no reason to ove because I guess you will want to merge `Apply` with `NDFrameApply` later?

Reply:
I've moved this to NDFrameApply